Owin.Security.Providers/Onshape/OnshapeAuthenticationOptions.cs

```diff
@@ -0,0 +1,97 @@
+        /// </remarks>
+        public ICertificateValidator BackchannelCertificateValidator { get; set; }
+
+        /// <summary>
+        ///     The HttpMessageHandler used to communicate with Onshape.
+        ///     This cannot be set at the same time as BackchannelCertificateValidator unless the value
+        ///     can be downcast to a WebRequestHandler.
+        /// </summary>
+        public HttpMessageHandler BackchannelHttpHandler { get; set; }
+
+        /// <summary>
+        ///     Gets or sets timeout value in milliseconds for back channel communications with Onshape.
+        /// </summary>
+        /// <value>
+        ///     The back channel timeout in milliseconds.
+        /// </value>
+        public TimeSpan BackchannelTimeout { get; set; }
+
+        /// <summary>
+        ///     The request path within the application's base path where the user-agent will be returned.
+        ///     The middleware will process this request when it arrives.
+        ///     Default value is "/signin-Onshape".
+        /// </summary>
+        public PathString CallbackPath { get; set; }
+
+        /// <summary>
+        ///     Get or sets the text that the user can display on a sign in user interface.
+        /// </summary>
+        public string Caption
+        {
+            get { return Description.Caption; }
+            set { Description.Caption = value; }
+        }
+
+        /// <summary>
+        ///     Gets or sets the Onshape supplied Application Key
+        /// </summary>
+        public string AppKey { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the Onshape supplied Application Secret
+        /// </summary>
+        public string AppSecret { get; set; }
+
+        /// <summary>
+        /// Hostname to use for the endpoints. Override this for connecting to the partner stack.
+        /// </summary>
+        public string Hostname { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the <see cref="IOnshapeAuthenticationProvider" /> used in the authentication events
+        /// </summary>
+        public IOnshapeAuthenticationProvider Provider { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the name of another authentication middleware which will be responsible for actually issuing a user
+        ///     <see cref="System.Security.Claims.ClaimsIdentity" />.
+        /// </summary>
+        public string SignInAsAuthenticationType { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the type used to secure data handled by the middleware.
+        /// </summary>
+        public ISecureDataFormat<AuthenticationProperties> StateDataFormat { get; set; }
+
+        /// <summary>
+        ///     Initializes a new <see cref="OnshapeAuthenticationOptions" />
+        /// </summary>
+        public OnshapeAuthenticationOptions()
+            : base("Onshape")
+        {
+            Caption = Constants.DefaultAuthenticationType;
```

Comment:
Please set default value for CallbackPath:

```
CallbackPath = new PathString("/signin-onshape");
```

Make sure it is all lower case